Copy Ability Adjustments + small fix

ethangarsia517 opened the pull request:
[assets.zip](https://github.com/HDR-Development/HewDraw-Remix/files/12502143/assets.zip)

Several Copy Abilities did not copy some of the new behavior that HDR added to some neutral specials in the game. This PR aims to fix that.

Should a character receive neutral special changes before this is pushed, they may or may not be adjusted for Kirby as well. Please let me know if there is a difference I may be missing here.

Lots of neutral specials also had some small frame data changes and have been matched. 
[Frame Data Changes.txt](https://github.com/HDR-Development/HewDraw-Remix/files/12447089/Frame.Data.Changes.txt)



**Main Changes:**

[+] Add Land Cancels for:
   - Fox
   - Bowser (auto transition into attack)
   - Falco
   - Zelda
   - Charizard (auto transition into attack)
   - Mr. Game & Watch
   - Dark Pit
   - Sora (Thundaga)
   - Banjo (on first shot)

[+] Add Fast Fall for lots of neutral specials in different statuses

[+] Add Aerial Drift for:
   - Link
   - Fox
   - Falco
   - Young Link
   - Toon Link
   - Mr. Game & Watch
   - Wolf
   - Bayonetta
   - Simon

[+] Add Airdodge Cancel on shoot for:
   - Diddy Kong
   - Wolf
   - Pac-Man
   - Mega Man
   - Sora (Firaga)

[+] Add Dash Cancel for:
   - Palutena (on hit)
   - Greninja (Max Charge)
   - Hero (Frizz, requires at least 13 mana and consumes 10)

[R] Add/Change Visual and Sound Effects for:
   - Mario
   - Luigi
   - Dr. Mario
   - Sonic
   - Palutena
   - Isabelle
   - King K. Rool
   - Mii Gunner

[!] Add PM-like Neutral Special Cancels

Bayonetta and Diddy Kong don't currently have animations for special canceling, so they are frame 1 for now.

**Copy Ability Changes:**

[!] PK Freeze: Kirby now harnesses the power of PSI, allowing him access to Offense Up.
   - Offense Up Changelog: [Offense Up Kirby.txt](https://github.com/HDR-Development/HewDraw-Remix/files/12447086/Offense.Up.Kirby.txt)


[+] Falcon Punch: can be turned around infinitely.
[+] Falcon Punch: adjusted KBG, BKB, and Damage output for aerial version to match Captain Falcon. This was an issue from vanilla Ultimate, now actually KOs.

[-] Giant Punch: Adjusted/removed hitboxes, damage, and KBG to match Donkey Kong.

[-] Bullet Climax: No longer gives Kirby Magic Series.

[/] Hadoken (Ken): Aerial version now shoots diagonally downwards.
(new animation)

[+] Plasma Breath: Can now go 💥 just like Ridley.
(new animation)

[!] Warlock Punch: Since Kirby loves to be in the air, he has learned how to Float just like Ganondorf instead.
(new animation)

[!] Bowser Firebreath: Kirby has mastered his firebreathing skills, and has learned how to shoot a powerful fireball just like Bowser.
(new animation)

[+] Straight Lunge: added jump and shield cancels.
(new animation)

[!] Ink Blaster: Repositioned hitbox, is not behind Kirby anymore.

**Other Changes:**

[+] Down Smash: Fixed a bug that caused the middle hitbox (ID2) damage to be decreased from 10 to 10 on late hit, when instead it was supposed to be one of the outer hitboxes (ID4) having its damage decreased from 14 to 10. All 4 outer hitboxes now have their damaged decreased to 10 on late hit.

[+] Hammer Flip: Can now drift and fastfall during the swing.


**Known Bugs:**
- Mii Gunner air to ground transition is broken when using the Heavy variant of Charge Shot.

Fixes #678

ethangarsia517 committed:
Base Commit

Mario Bros, and Isabelle effects, most cancel code, G&W fix, wario backwards throw

ethangarsia517 committed:
Base Commit

Mario Bros, and Isabelle effects, most cancel code, G&W fix, wario backwards throw

ethangarsia517 committed:
Merge branch 'HDR-Development:dev' into copyabilityfix

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
Merge branch 'copyabilityfix' of https://github.com/ethangarsia517/HewDraw-Remix into copyabilityfix

ethangarsia517 committed:
fix luigi gfx, add aerial drifts

ethangarsia517 committed:
fix luigi gfx, add aerial drifts

ethangarsia517 committed:
fix bullet arts aerial drift plus

ethangarsia517 committed:
fix bullet arts aerial drift plus

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
add pm-like neutralb cancels

bayonetta, diddy kong, and little mac need special animations so they are currently f1

ethangarsia517 committed:
add pm-like neutralb cancels

bayonetta, diddy kong, and little mac need special animations so they are currently f1

ethangarsia517 committed:
even more neutral b tech + zelda touchup + master pmlike cancel

because I rarely play against any of these so i didnt know lmao (does anyone even play diddy kong)

ethangarsia517 committed:
even more neutral b tech + zelda touchup + master pmlike cancel

because I rarely play against any of these so i didnt know lmao (does anyone even play diddy kong)

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
match frame data + fix incineroar&luigi slipoff

wow that's a lot of missed details, maybe I dont need the gaogaen install thingy tho

ethangarsia517 committed:
match frame data + fix incineroar&luigi slipoff

wow that's a lot of missed details, maybe I dont need the gaogaen install thingy tho

ethangarsia517 committed:
add/fix frizz dash cancel

ethangarsia517 committed:
add/fix frizz dash cancel

ethangarsia517 committed:
add warlock punch turnarounds + miigunner fixes

broken copy abilities became simply out of scope of this PR.

ethangarsia517 committed:
add warlock punch turnarounds + miigunner fixes

broken copy abilities became simply out of scope of this PR.

SkewedAskew reviewed (changes requested):
One small thing:
- Nayru's Love should have 7 frames of landing lag as opposed to being a true land cancel. Her land cancel in opff sets the animation to a specific frame on landing, use that same logic

ethangarsia517 committed:
zelda landing lag fix

ethangarsia517 committed:
zelda landing lag fix

ethangarsia517 commented:
> One small thing:
> 
> * Nayru's Love should have 7 frames of landing lag as opposed to being a true land cancel. Her land cancel in opff sets the animation to a specific frame on landing, use that same logic

I think this should do the trick (matches Zelda's opff now.) Forgot about this one, it was before I found out that character specific statuses like these work for Kirby as well.

https://github.com/HDR-Development/HewDraw-Remix/pull/1821/commits/bf62c46a65cdd45ab1fa90aab9798d24d7735e91

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
give fastfalls + disable previous fastfall code

253 more statuses of potential fastfalling. if spacies/gamewatch don't function, will re-enable

ethangarsia517 committed:
give fastfalls + disable previous fastfall code

253 more statuses of potential fastfalling. if spacies/gamewatch don't function, will re-enable

ethangarsia517 committed:
whoops forgot pichu

i have no idea how this happened

ethangarsia517 committed:
whoops forgot pichu

i have no idea how this happened

ethangarsia517 committed:
fix spacies+gamewatch fastfall, remove old fastfall code for them

PR is now complete.

ethangarsia517 committed:
fix spacies+gamewatch fastfall, remove old fastfall code for them

PR is now complete.

SkewedAskew commented on fighters/kirby/src/opff.rs:
```diff
@@ -320,6 +899,260 @@ unsafe fn fastfall_specials(fighter: &mut L2CFighterCommon) {
         *FIGHTER_KIRBY_STATUS_KIND_SPECIAL_S_FALL,
         *FIGHTER_KIRBY_STATUS_KIND_SPECIAL_S_JUMP,
         *FIGHTER_KIRBY_STATUS_KIND_SPECIAL_S_ATTACK,
+        //copy abilities
```
Instead of checking every Kirby special_n status individually, try something like this:
```rs
let status = StatusModule::status_kind(fighter.module_accessor);
if (0x206..0x37c).contains(&status) {
  // do the thing
}
```
These two values should encompass the range of the Copy Ability statuses

ethangarsia517 committed:
simplify 253 fastfall status into 2 lines

whew thats better

ethangarsia517 committed:
simplify 253 fastfall status into 2 lines

whew thats better

SkewedAskew reviewed (dismissed):
Excellent work, very much appreciated

zrksyd committed:
Kirby with Ken's aerial Hadouken

+ extra checks for Ryu and Ken for Kirby to accidentally not gain meter

Note: Kirby can perform true Hadouken with A and B but Ken can only with B.

zrksyd committed:
Fixing rebasing issues

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
Merge branch 'copyabilityfix' into copyabilityfix

ethangarsia517 committed:
Merge pull request #1 from zrksyd/copyabilityfix

Kirby with Ken's aerial Hadouken

zrksyd committed:
Ridley Explode for Kirby

No animation included

ethangarsia517 committed:
Merge pull request #2 from zrksyd/copyabilityfix

Ridley Explode for Kirby

ethangarsia517 committed:
oops ridley

zrksyd committed:
Kirby with Ganon's float

Note: Kirby cannot up B after Ganon float until landing. Kriby cannot Ganon float after a successful Cutter Dash.

zrksyd committed:
Kirby copy abliity Roy, Ridley, Sonic

zrksyd committed:
Fix Kirby neutral B / up B in same airtime bug

zrksyd committed:
Move Kirby copy ACMD to new file

Alphabetized scripts + Kirby Isabelle Neutral B Expression

zrksyd committed:
Kirby Ganon Float sound / expression

zrksyd committed:
New Bowser Neutral B for Kirby

ethangarsia517 committed:
miigunner & mariod redo, remove gamewatch opff

ethangarsia517 committed:
miigunner & mariod redo, remove gamewatch opff

zrksyd committed:
Straight Lunge cancel for Kirby

ethangarsia517 committed:
Merge branch 'copyabilityfix' of https://github.com/ethangarsia517/HewDraw-Remix into copyabilityfix

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

zrksyd committed:
Lucas' Offense Up for Kirby

Smash attack "buffed" hitboxes are identical to base besides electric properties

zrksyd committed:
Renaming Bowser max cooldown variable

zrksyd committed:
Kirby Bowser Fireball Expresssion

ethangarsia517 committed:
krool, corrin, palutena, inkling, lucas adjustments, offense up buffs

Effects for krool and palutena and lucas, frame data for corrin and palutena, hitbox adjustment for inkling, offense up buffs

zrksyd committed:
Kirby Offense Up Smash Sound adjustment

zrksyd committed:
Kirby Hadouken with only B

zrksyd committed:
Kirby Palutena Rumble

zrksyd committed:
Peanut Popgun Cancel for Kirby

ethangarsia517 committed:
fix float

changes the values in const.rs to be unique.

ethangarsia517 committed:
Merge branch 'copyabilityfix' of https://github.com/ethangarsia517/HewDraw-Remix into copyabilityfix

zrksyd committed:
countdown int fix

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
captain and donkey acmd fix

fixes aerial falcon punch being extremely weaker from vanilla, because wtf ninty. also brings giant punch up to date with current HDR original

ethangarsia517 committed:
Revert "Peanut Popgun Cancel for Kirby"

This reverts commit 42b6b8ac9c835e6fe97ba60493c6038511de568b.

ethangarsia517 committed:
Merge remote-tracking branch 'upstream/dev' into copyabilityfix

ethangarsia517 committed:
nightly changes

staying up to date with the latest stuff that has been merged

ethangarsia517 committed:
hammer swing fastfall drift

ethangarsia517 committed:
Merge branch 'dev' into copyabilityfix

SkewedAskew reviewed (approved):
We'll need to keep a close eye out for any potential bugs, but great work
